app/models/user.rb

```diff
@@ -51,4 +56,23 @@ def toggle_admin!
     team = Registry.first.global_namespace.team
     admin ? team.owners << self : team.owners.delete(self)
   end
+
+  ##
+  # Disabling users.
+
+  # Disable this user and remove all its associations with teams.
+  def disable!
+    return unless update_attributes(enabled: false)
+    TeamUser.where(user: id).delete_all
```

Comment:
Ok makes sense. So the only thing to do would be to hide disabled users in views.